### PR TITLE
Update French translation

### PR DESCRIPTION
**Pull request by rdoursenaud**
Also added non breakable spaces before double punctuation signs according to French typographic rules.

### CODE REVIEW

File: src/translations/qjackctl_fr.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <message>
         <location filename="../qjackctlInterfaceComboBox.cpp" line="157"/>
         <source>Please wait, PortAudio is probing audio hardware.</source>
-        <translation>Merci de patienter, PortAudio détecte le matériel audio.</translation>
+        <translation>PortAudio détecte le matériel audio, merci de patienter.</translation>
```
Comment (trebmuh) [inline src/translations/qjackctl_fr.ts:14]: Why this? It doesn't add anything and it reverse the original flow.

File: src/translations/qjackctl_fr.ts [hunk 2]
```diff
@@ -567,12 +567,12 @@ de toutes les applications clientes.
     <message>
         <location filename="../qjackctlGraphForm.ui" line="199"/>
         <source>Ctrl+C</source>
-        <translation type="unfinished"></translation>
+        <translation>Ctrl+C</translation>
```
Comment (trebmuh) [inline src/translations/qjackctl_fr.ts:570]: There's no need to translate this, as the translation system will automatically take the English version when compiling. What's more, there's a risk of errors or omissions in the long term (especially if the English string is changed).

File: src/translations/qjackctl_fr.ts [hunk 3]
```diff
@@ -567,12 +567,12 @@ de toutes les applications clientes.
...
     </message>
     <message>
         <location filename="../qjackctlGraphForm.ui" line="219"/>
         <source>Ctrl+D</source>
-        <translation type="unfinished"></translation>
+        <translation>Ctrl+D</translation>
```
Comment (trebmuh) [inline src/translations/qjackctl_fr.ts:575]: same here

**Review by trebmuh — CHANGES_REQUESTED** [review]

**Comment by trebmuh** [comment]
I'm unsure about the "non breakable spaces". I'll let Rui decide on those.

File: src/translations/qjackctl_fr.ts (hunk 1 above)
Comment (rdoursenaud) [reply src/translations/qjackctl_fr.ts:14]: Because that's the correct wording order for french IMHO.

File: src/translations/qjackctl_fr.ts (hunk 2 above)
Comment (rdoursenaud) [reply src/translations/qjackctl_fr.ts:570]: Well then we should review all the other shortcuts which have been translated.
Moreover, the key is currently the same as the source version but is actually determined by the "&<letter>" syntax in the menu text translation string.
A better fix would be to grab this programmatically and not rely on a separate string.

**Comment by rdoursenaud** [comment]
Thanks for your review. I've added my though process in the comments. Feel free to discuss.

**Comment by rncbc** [comment]
what's the matter about the "non breakable spaces" ? is that really necessary to a space (breakable or not) to precede a semi-colon? i guess it would make everybody happy if there was none, but that's probably just a non-French speaking here ;)

File: src/translations/qjackctl_fr.ts (hunk 1 above)
Comment (trebmuh) [reply src/translations/qjackctl_fr.ts:14]: Both ways are correct as far as I know.

File: src/translations/qjackctl_fr.ts (hunk 2 above)
Comment (trebmuh) [reply src/translations/qjackctl_fr.ts:570]: > Well then we should review all the other shortcuts which have been translated.

Agreed.

**Comment by trebmuh** [comment]
>  is that really necessary to a space (breakable or not) to precede a semi-colon?

In French, yes.

File: src/translations/qjackctl_fr.ts (hunk 2 above)
Comment (rdoursenaud) [reply src/translations/qjackctl_fr.ts]: I've removed the translations where appropriate.

**Comment by rncbc** [comment]
@trebmuh , @rdoursenaud : is it good to go now?

v0.9.11 is nigh ;)

File: src/translations/qjackctl_fr.ts (hunk 1 above)
Comment (trebmuh) [reply src/translations/qjackctl_fr.ts:14]: I'd like this change to be reverted if you don't mind.

File: src/translations/qjackctl_fr.ts (hunk 2 above)
Comment (trebmuh) [reply src/translations/qjackctl_fr.ts]: Thanks for that.

**Comment by trebmuh** [comment]
With one more change to be reverted, it looks OK to me.

I've got one question that I don't have an answer and I'm relying on both of you for: let's take an example:
```
    <message>
        <source>Alt+A</source>
        <translatorcomment>Do not translate</translatorcomment>
        <translation></translation>
    </message>
```

What will be the outcome of this example:
* will the French translation fallback to use "Alt+A" as well ?
* or will the French translation use "_nothing_" because there is nothing between `<translation>` and `</translation>` and here is no "unfinished" tag?

File: src/translations/qjackctl_fr.ts (hunk 1 above)
Comment (rdoursenaud) [reply src/translations/qjackctl_fr.ts:14]: On the way.

**Comment by rdoursenaud** [comment]
> * will the French translation fallback to use "Alt+A" as well ?
> * or will the French translation use "_nothing_" because there is nothing between `<translation>` and `</translation>` and here is no "unfinished" tag?

It's exactly because this behavior is non-obvious that I made sure to fill every translation field even if the translation was identical to the "C" language string.
I don't have the answer and can't test right now since I'm away from my Linux workstation and don't want to bother compiling under Windows.

Please let me know if I need to change anything.

**Comment by rncbc** [comment]
>* will the French translation fallback to use "Alt+A" as well ?

yes

>* or will the French translation use "nothing" because there is nothing between <translation> and </translation> and here is no "unfinished" tag?

no. but in doubt leave the "unfinished" there as it was.

**Comment by rdoursenaud** [comment]
> no. but in doubt leave the "unfinished" there as it was.

This makes life harder as a translator :(
I rely on this information to know which string has been updated and needs my attention.
Plus similar untranslated strings weren't marked unfinished before.

**Comment by rncbc** [comment]
if you touched it, just leave as it was before

anyway should pay attention to the:
`<translatorcomment>Do not translate</translatorcomment>`

**Pull request closed without merging**